es/es-server/src/main/java/org/elasticsearch/repositories/fs/FsRepository.java

```diff
@@ -62,6 +63,14 @@
     public static final Setting<ByteSizeValue> REPOSITORIES_CHUNK_SIZE_SETTING = Setting.byteSizeSetting("repositories.fs.chunk_size",
         new ByteSizeValue(Long.MAX_VALUE), new ByteSizeValue(5), new ByteSizeValue(Long.MAX_VALUE), Property.NodeScope);
 
+    public static List<Setting> mandatorySettingsToValidate() {
+        return List.of(LOCATION_SETTING);
+    }
+
+    public static List<Setting> optionalSettingsToValidate() {
```

Comment:
```suggestion
    public static List<Setting> optionalSettings() {
```

The `toValidate` part implies a very specific use-case, that is also kind of redundant.